### DISCUSSION TITLE
Resource Redirection

### DISCUSSION THREAD

**Discussion by WildByDesign** [ramensoftware/windhawk-mods#633] [General]
@m417z

You have released the Icon Resource Redirect mod recently and it got me wondering. Could such a resource redirection technique also work with msstyles theme resources?

For example, there is an issue with programs that use `win32-darkmode` (Notepad++, Explorer++, LibreOffice, etc.) on Windows 11 specifically where the toolbar icons have a light blue highlighting that looks terrible and Microsoft still hasn't fixed.

Link: https://github.com/derceg/explorerplusplus/issues/115#issuecomment-1598660527

If that one single image resource could be redirected/replaced on-the-fly, that would be incredible.

Thank you for the work that you do. I've watched Windhawk since the beginning and it's fantastic to see the Windhawk community blossoming now. Cheers!

**Comment by m417z** [comment]
Hi, thanks for the feedback about Windhawk, I'm glad that you like it!

> Could such a resource redirection technique also work with msstyles theme resources?

From a quick test, I'm afraid that the answer is no. I switched my Notepad++ to dark mode, and it seems that the `aero.msstyles` file isn't being accessed. I assume that the resources are loaded by the system in a more centralized way and are shared between all apps.

I'm not that familiar with the theme system internals in Windows, but it's possible that a solution like [SecureUxTheme](https://github.com/namazso/SecureUxTheme) (which I haven't personally used) is the easiest way to make such a resource replacement.

**Reply by m417z** [reply]
Going over the readme of SecureUxTheme, I stumbled upon this FAQ entry:

> ### **Q: Can you make themes per-program?**
>
> A: Unfortunately, this is [close to impossible](https://github.com/namazso/SecureUxTheme/issues/9#issuecomment-611897882).

I'm not sure at which locations the button bitmap is retrieved, but the above analysis suggests that it might be infeasible to hook them all. Maybe @namazso has some quick insight on this.
